jetty-http2/http2-hpack/src/main/java/org/eclipse/jetty/http2/hpack/HpackDecoder.java

```diff
@@ -267,14 +274,61 @@ public MetaData decode(ByteBuffer buffer) throws HpackException.SessionException
         return _builder.build();
     }
 
-    public static String toASCIIString(ByteBuffer buffer, int length)
+    private int integerDecode(ByteBuffer buffer, int prefix) throws HpackException.CompressionException
     {
-        StringBuilder builder = new StringBuilder(length);
+        try
+        {
+            if (prefix != 8)
+                buffer.position(buffer.position() - 1);
+
+            _integerParser.setPrefix(prefix);
+            int decodedInt = _integerParser.decodeInt(buffer);
+            if (decodedInt < 0)
+                throw new EncodingException("invalid integer encoding");
+            return decodedInt;
+        }
+        catch (EncodingException e)
+        {
+            HpackException.CompressionException compressionException = new HpackException.CompressionException(e.getMessage());
+            compressionException.initCause(e);
+            throw compressionException;
+        }
+        finally
+        {
+            _integerParser.reset();
+        }
+    }
+
+    private String huffmanDecode(ByteBuffer buffer, int length) throws HpackException.CompressionException
+    {
+        try
+        {
+            _huffmanDecoder.setLength(length);
+            String decoded = _huffmanDecoder.decode(buffer);
+            if (decoded == null)
+                throw new HpackException.CompressionException("invalid string encoding");
+            return decoded;
+        }
+        catch (EncodingException e)
+        {
+            HpackException.CompressionException compressionException = new HpackException.CompressionException(e.getMessage());
+            compressionException.initCause(e);
+            throw compressionException;
+        }
+        finally
+        {
+            _huffmanDecoder.reset();
+        }
+    }
+
+    public static String toISO8859String(ByteBuffer buffer, int length)
+    {
+        CharsetStringBuilder.Iso8859StringBuilder builder = new CharsetStringBuilder.Iso8859StringBuilder();
         for (int i = 0; i < length; ++i)
         {
             builder.append((char)(0x7F & buffer.get()));
```

Comment:
Change it to `buffer.get() & 0x7F`.
Also feels a bit weird... are you sure this is the right logic? If so, can you comment why, maybe reference the spec if it comes from some RFC?

Reply:
This drops the high-bit, that seems like a bug.

Reply:
Yeah probably because this was originally for ASCII, I will remove the 0x7F and use the `HttpTokens.sanitizeFieldVchar` instead.